### PR TITLE
Find CompositeOperator instance in Image#compose instead of generate it

### LINKED ISSUE #586: Image#compose might cause infinity loop
If call Image#compose method, it create new instance and insert it into Magick::CompositeOperator.
Then if refer constant of Magick::CompositeOperator using `Magick::CompositeOperator.values`, it causes infinity loop.

* Reproduce code
```ruby
require 'rmagick'

img = Magick::Image.new(10, 10)
Magick::CompositeOperator.values do |op|
  p op
  p "#{op}, #{img.compose}"
end
```

* Result
```
$ ruby test.rb
AddCompositeOp=2
"AddCompositeOp, OverCompositeOp"
AtopCompositeOp=3
"AtopCompositeOp, OverCompositeOp"
BlendCompositeOp=4
"BlendCompositeOp, OverCompositeOp"
BlurCompositeOp=57
...

OverCompositeOp=40
"OverCompositeOp, OverCompositeOp"
OverCompositeOp=40
"OverCompositeOp, OverCompositeOp"
OverCompositeOp=40
"OverCompositeOp, OverCompositeOp"
OverCompositeOp=40
"OverCompositeOp, OverCompositeOp"
OverCompositeOp=40
"OverCompositeOp, OverCompositeOp"
...

```

### PR DESCRIPTION
**Pull request by Watson1978**
Fix #586

Find and return already created instance of CompositeOperator.

### CODE REVIEW

File: ext/RMagick/rmenum.c [hunk 1]
```diff
@@ -361,6 +361,38 @@ Enum_type_values(VALUE class)
...
+    VALUE enumerators;
+    MagickEnum *magick_enum;
+    int x;
+
+    enumerators = rb_cv_get(class, ENUMERATORS_CLASS_VAR);
+    enumerators = rm_check_ary_type(enumerators);
+
+    for (x = 0; x < RARRAY_LEN(enumerators); x++)
+    {
+       VALUE e = rb_ary_entry(enumerators, x);
```
Comment (dlemstra) [inline ext/RMagick/rmenum.c]: Maybe you could use a more descriptive name here instead of just `e`?

**Comment by Watson1978** [comment]
> I am having trouble to understand why this fails but doesn't this mean we have this problem for all the other enumerations that are also created like this?

Of course. I just fix step by step to make review easily.

**Comment by dlemstra** [comment]
I was not sure if that was also the case. Does that mean we should remove all the `_new` and `_name` methods for all the enumerations? I don't mind helping you out with this if you could use some help.

**Comment by Watson1978** [comment]
> I was not sure if that was also the case. Does that mean we should remove all the _new and _name methods for all the enumerations? I don't mind helping you out with this if you could use some help.

I've not confirmed other case yet. But, I believe the same bug is existed with same case.

File: ext/RMagick/rmenum.c (hunk 1 above)
Comment (Watson1978) [reply ext/RMagick/rmenum.c]: renamed :)

Comment (dlemstra) [reply ext/RMagick/rmenum.c]: Thanks!

**Comment by dlemstra** [comment]
I can confirm we also have this bug for other enumerations. This also creates a loop of doom.

```ruby
require 'rmagick'

img = Magick::Image.new(10, 10)
Magick::ClassType.values do |op|
  p op
  p "#{op}, #{img.class_type}"
end
```

**Comment by Watson1978** [comment]
@dlemstra Please file the issue :)

**Comment by dlemstra** [comment]
Maybe we should create one issue and use that for all the enumerations that should be refactored?

**Comment by Watson1978** [comment]
i will leave it to you :)

**Comment by Watson1978** [comment]
Thanks 🥳

**Pull request merged**